src/core/src/test/java/org/locationtech/geogig/data/EPSGBoundsCalcTest.java

```diff
@@ -0,0 +1,105 @@
+/*
+ *  Copyright (c) 2017 Boundless and others.
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Distribution License v1.0
+ *  which accompanies this distribution, and is available at
+ *  https://www.eclipse.org/org/documents/edl-v10.html
+ *
+ *  Contributors:
+ *  Morgan Thompson (Boundless) - initial implementation
+ *  Alex Goudine (Boundless)
+ */
+
+package org.locationtech.geogig.data;
+
+import com.google.common.base.Optional;
+import com.vividsolutions.jts.geom.Envelope;
+import org.junit.Test;
+import org.locationtech.geogig.model.RevFeatureType;
+import org.locationtech.geogig.plumbing.ResolveFeatureType;
+import org.locationtech.geogig.porcelain.CommitOp;
+import org.locationtech.geogig.repository.NodeRef;
+import org.locationtech.geogig.test.integration.RepositoryTestCase;
+import org.opengis.feature.type.GeometryDescriptor;
+import org.opengis.feature.type.PropertyDescriptor;
+import org.opengis.referencing.ReferenceIdentifier;
+
+import java.util.List;
+
+/**
+ * Created by mthompson on 2017-01-12.
+ */
+public class EPSGBoundsCalcTest extends RepositoryTestCase {
+
+    @Override
+    protected void setUpInternal() throws Exception {
+        injector.configDatabase().put("user.name", "mthompson");
+        injector.configDatabase().put("user.email", "mthompson@boundlessgeo.com");
+    }
+
+    @Test
+    public void metadataIdTest() throws Exception {
+        Envelope bounds = null;
+        ReferenceIdentifier code = null;
+
+        insertAndAdd(points1);
+        geogig.command(CommitOp.class).setMessage("Commit1").call();
+
+        Optional<RevFeatureType> featureType = geogig.command(ResolveFeatureType.class)
+            .setRefSpec("WORK_HEAD:" + NodeRef.appendChild(pointsName, idP1)).call();
+
+        List<PropertyDescriptor> descList = featureType.get().descriptors().asList();
```

Comment:
You can refactor this to the EPSGBoundsCalc

public Envelope getCRSBounds(GeometryDescriptor gd)